metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1999,7 +1999,11 @@ class MetalsLanguageServer(
           val uri = textDocumentParams.getTextDocument().getUri()
           for {
             edits <- compilers.insertInferredType(textDocumentParams, token)
-            if (!edits.isEmpty())
+            _ = if (edits.isEmpty())
+              scribe.error(
+                s"Could not infer type at ${textDocumentParams.getPosition()} in file\n${buffers
+                    .get(uri.toAbsolutePath)}"
+              )
```

Comment:
What do you think about creating wrapper on `log if condition is met`
```scala
package object internal {
  private def scribeLogWhen(level: Level, condition: Boolean, msg: String)(implicit
      pkg: sourcecode.Pkg,
      fileName: sourcecode.FileName,
      name: sourcecode.Name,
      line: sourcecode.Line,
      mdc: MDC,
  ): Unit =
    if (condition)
      scribe.log(level, mdc, msg)

  def scribeLogError(condition: Boolean)(msg: String)(implicit
      pkg: sourcecode.Pkg,
      fileName: sourcecode.FileName,
      name: sourcecode.Name,
      line: sourcecode.Line,
      mdc: MDC,
  ): Unit = scribeLogWhen(Level.Error, condition, msg)
}
```
Then we could save on all those ifs :D